components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/api/model/EndpointConfig.java

```diff
@@ -45,13 +52,25 @@ public Authentication getAuthentication() {
         return authentication;
     }
 
+    public List<String> getAllowedHeaders() {
+
+        return (allowedHeaders == null) ? null : Collections.unmodifiableList(allowedHeaders);
```

Comment:
I think it's better to have two distinct scenarios to denote when allowedHeaders are not configured and configured.
Like,
No allowed headers -> empty List
Configured allowed header -> List

or 

No allowed headers -> null
Configured allowed header -> List

Handling null, emptyList and a non empty List as values for the variable can be problematic in some scenarios.